app/src/main/java/ar/scacchipa/twittercloneapp/domain/AuthorizationUseCase.kt

```diff
@@ -0,0 +1,45 @@
+package ar.scacchipa.twittercloneapp.domain
+
+import ar.scacchipa.twittercloneapp.datasource.UserAccessToken
+import ar.scacchipa.twittercloneapp.datasource.provideAuthSourceDateApi
+import ar.scacchipa.twittercloneapp.datasource.provideRetrofit
+import ar.scacchipa.twittercloneapp.repository.AuthorizationRepository
+import ar.scacchipa.twittercloneapp.repository.IAuthorizationRepository
+
+class AuthorizationUseCase(
+    private val repository: IAuthorizationRepository =
+        AuthorizationRepository(provideAuthSourceDateApi(provideRetrofit()))
+) {
+    suspend fun generateAccessToken(
+        transitoryToken: String,
+        grant_type: String = "authorization_code",
```

Comment:
not the right way of naming variables , should be something like grantType.

Reply:
The use case was splitted into AuthorizationUseCase and ConsumableAuthUseCase.
Both classes have their invoke operators